content/blogposts/2023/blah.md

```diff
@@ -0,0 +1,151 @@
+</a>
+
+Luckily, embeddings search alone often includes a lot of the critical context to make magical-seeming inferences. In this example, the _description_ of the cursor is correct after reading the related GraphQL schema file, as well as some other files in the same package which _by happenstance_ also make use the cursor object. This is why the `Phase` field is known to have the values `"local"` and `"remote"`. All of this is spot-on, although oddly the value `"done"` is not explicitly listed as a possible phase even though it visible in the source of the screenshot above.
+
+The remainder of the response slips into a hallucination. While `RepositoryID` and `Commit` do not actually exist in the cursor, similar fields do exist on a sibling `RequestArgs` struct that's passed around many of the same code paths. This is likely a source of the confusion in the response. The three remaining fields `Definitions`, `References`, and `Prototypes` do not exist on the cursor, but they do make remarkable sense given the context of the code in which this cursor is being used. These fields could absolutely be a valid alternate design for an API in the same domain.
+
+Generative AI is simply a machine that finishes a pattern. If sufficient examples are not available to narrow which pattern to follow, it seems that *anything* can come shooting out. An AI coding assistant that always response with the know-it-all answer of _"you could've designed it **this other way**"_ is not only unhelpful, but enraging to hear. Especially if you've previously fallen for AI refactor suggestions and _the other way_ is how it was originally written.
+
+We'll sufficiently narrow down the pattern-space by supplying additional (and highly relevant) code, just as we did above. With enough context, answers will adapt themselves to the environment (as we'll see below) rather than making leaping inferrences about the environment to support its output (as we saw above).
+
+<a href="https://storage.googleapis.com/sourcegraph-assets/blog/scip-powered-cody/cursor-scip.png" target="_blank">
+    <Figure
+        src="https://storage.googleapis.com/sourcegraph-assets/blog/scip-powered-cody/cursor-scip.png"
+        alt="SCIP-powered Cody inferring the correct struct tags."
+        caption="SCIP-powered Cody inferring the correct struct tags." />
+</a>
+
+The skeptical reader may have noticed a heavy similarity between the first two examples - both questions are basically using an AI layer to emulate a fairly simple jump to definition action. Asking these questions in natural language probably takes more time than hovering over the symbol to peek at its definition. These examples are _highly illustrative_ but also very myopic with respect to what's possible with this technique.
+
+Continue reading to see that precise code intelligence isn't a one-trick-pony source of context.
+
+### Deeplying understanding code
```

Comment:
I really like this example. It is, as you note, simplified. However, it also illustrates the value well.